bin/varnishd/cache/cache_vrt.c

```diff
@@ -721,8 +721,11 @@ VRT_TIME_string(VRT_CTX, VCL_TIME t)
 
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
 	p = WS_Alloc(ctx->ws, VTIM_FORMAT_SIZE);
-	if (p != NULL)
+	if (p != NULL) {
 		VTIM_format(t, p);
+		if (*p == '\0')
+			p = NULL;
+	}
```

Comment:
We are wasting `VTIM_FORMAT_SIZE` bytes of workspace on the failure condition. Either we `WS_Snapshot()` then `WS_Reset()` on error, or `WS_ReserveSize()` then `WS_Release()` accordingly.

Reply:
This is a very exceptional case, so it wouldn't be a big deal, but added for completeness.